File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int n) {
    int answer = 0;
    
    vector<bool> num(n, true);
    num[0] = false;
    
    for(int i = 2; i < n + 1; i++)
    {
        int j = 2;
        
        while(i * j <= n)
        {
            num[i*j - 1] = false;
            j++;
        }
    }
    
    answer = count(num.begin(), num.end(), true);
    
    return answer;
}

File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.py
def solution(n):
    answer = 0
    
    a = [True] * (n+1)
    a[0] = False
    a[1] = False
    
    for k in range (2,n+1):
        if a[k] == True:
            i=k
            while i<=(n-k):
                a[k+i]=False
                i=i+k
                
        else:
            a[k] = False
            
    count_true = sum(a)
    
    return count_true